File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l = 0
        r = len(s) - 1
        
        while l < r:
            # Move left pointer to the right if not alphanumeric
            while l < r and not s[l].isalnum():
                l += 1
            # Move right pointer to the left if not alphanumeric
            while l < r and not s[r].isalnum():
                r -= 1
            # Check if characters at both pointers are equal (case insensitive)
            if s[l].lower() != s[r].lower():
                return False
            # Move pointers inward
            l += 1
            r -= 1
        
        return True

File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
    let alpha = "abcdefghijklmnopqrstuvwxyz1234567890"
    let l = 0
    let r = s.length - 1

    while (l < r) {
        let left = s[l].toLowerCase()
        let right = s[r].toLowerCase()
        if (!alpha.includes(left)) l++
        else if (!alpha.includes(right)) r--

        else {
            if (left != right) return false
            l++
            r--
        }
    }
    return true;
};